src/main/java/org/tarantool/jdbc/SQLConnection.java

```diff
@@ -296,21 +324,43 @@ public int getHoldability() throws SQLException {
 
     @Override
     public Savepoint setSavepoint() throws SQLException {
+        checkNotClosed();
+        if (getAutoCommit()) {
+            throw new SQLNonTransientException(
+                    "Cannot set a savepoint when auto-commit is enabled.",
```

Comment:
> if setSavepoint is invoked outside of an active transaction, a transaction will be started at this newly created savepoint.

https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html#setSavepoint--

The same for setSavepoint(name).

Reply:
Can you clarify your comment a bit?
This highlighted block follows another spec requirement: 
SQLException - if a database access error occurs, this method is called while participating in a distributed transaction, this method is called on a closed connection or **this Connection object is currently in auto-commit mode**

Reply:
I understood now: setSavepoint() cannot be called in auto-commit mode, but when the mode is disabled it is not necessary to start a transaction explicitly somehow.